sip-core/src/main/java/de/ikor/sip/foundation/core/premiumsupport/registration/AdapterRegistration.java

```diff
@@ -0,0 +1,50 @@
+package de.ikor.sip.foundation.core.premiumsupport.registration;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.DisposableBean;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.scheduling.annotation.ScheduledAnnotationBeanPostProcessor;
+import org.springframework.scheduling.config.ScheduledTask;
+import org.springframework.stereotype.Component;
+
+/**
+ * This class handles the SIP Backend registration of an adapter instance. It also scans whole
+ * premiumsupport.registration package, and depends on sip.core.backend-registration.enabled
+ * configuration.
+ */
+@Slf4j
+@Component
+@ComponentScan()
+@RequiredArgsConstructor
+@ConditionalOnProperty(name = "sip.core.backend-registration.enabled", havingValue = "true")
+public class AdapterRegistration implements InitializingBean, DisposableBean {
+  private final ScheduledAnnotationBeanPostProcessor processor;
+  private final SIPRegistrationClient registrationClient;
+
+  /**
+   * Is executed after the application has been started and its properties have been set, and
+   * scheduled for further invocations based on sip.core.backend-registration.interval property.
+   */
+  @Override
+  @Scheduled(fixedDelayString = "${sip.core.backend-registration.interval}")
+  public void afterPropertiesSet() {
+    this.registrationClient.registerAdapter();
+  }
```

Comment:
We already discussed this. registerAdapter implies having a register/health in a loop/unregister flow. I don't actually know of a better name, just this one was kind of misleading. Maybe changing it to sendTelemetryData?